### DISCUSSION TITLE
CopyWith set null?

### DISCUSSION THREAD

**Discussion by jopmiddelkamp** [rrousselGit/freezed#956] [General]
How does the copyWith method understand which argument has been set to null and, as a result, knows to reset that property?

**Comment by rrousselGit** [comment]
It's relying on default values.

```dart
copyWith({param = sentinel}) {
  if (param!= sentinel) 
}
```

**Comment by jopmiddelkamp** [comment]
Thanks for your quick reply! I think I see what you mean now!
